msgp/read_bytes.go

```diff
@@ -395,6 +395,52 @@ func ReadInt64Bytes(b []byte) (i int64, o []byte, err error) {
 		o = b[9:]
 		return
 
+	case muint8:
+		if l < 2 {
+			err = ErrShortBytes
+			return
+		}
+		i = int64(getMuint8(b))
+		o = b[2:]
+		return
+
+	case muint16:
+		if l < 3 {
+			err = ErrShortBytes
+			return
+		}
+		i = int64(getMuint16(b))
+		o = b[3:]
+		return
+
+	case muint32:
+		if l < 5 {
+			err = ErrShortBytes
+			return
+		}
+		i = int64(getMuint32(b))
+		o = b[5:]
+		return
+
+	case muint64:
+		if l < 9 {
+			err = ErrShortBytes
+			return
+		}
+		u := getMuint64(b)
+		if u > math.MaxInt64 {
+			err = badPrefix(IntType, lead)
+		}
+		i = int64(u)
+		o = b[9:]
+		return
+
+	case mfloat64:
+		var f float64
+		f, o, err = ReadFloat64Bytes(b)
+		i = int64(f)
```

Comment:
This seems like it could lose precision, but we're inside a ReadInt64 function, so this is the best we can do?